src/contracts/BCoWFactory.sol

```diff
@@ -25,11 +25,17 @@ contract BCoWFactory is BFactory {
    * @dev Deploys a BCoWPool instead of a regular BPool, maintains the interface
    * to minimize required changes to existing tooling
    */
-  function newBPool() external override returns (IBPool _pool) {
+  function newBPool() external override(BFactory, IBFactory) returns (IBPool _pool) {
     IBPool bpool = new BCoWPool(SOLUTION_SETTLER, APP_DATA);
     _isBPool[address(bpool)] = true;
     emit LOG_NEW_POOL(msg.sender, address(bpool));
     bpool.setController(msg.sender);
     return bpool;
   }
+
+  /// @inheritdoc IBCoWFactory
+  function logBCoWPool() external {
+    if (!_isBPool[msg.sender]) revert BCoWFactory_NotValidBCoWPool();
+    emit COWAMMPoolCreated(msg.sender);
```

Comment:
why is the event emitted when the pool is finalized instead of when it is created?

Reply:
Because when the pool is finalized is fully working and open. From conversations with Weiser I assume that the finalized pools should be easily indexed scanning only this event, and no extra off-chain logic.

Reply:
correct!